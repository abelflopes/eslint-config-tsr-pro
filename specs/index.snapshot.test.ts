import baseConfig from "../src";

describe("eslint config", () => {
  it("configuration remains as expected", () => {
    expect(baseConfig).toMatchSnapshot();
  });
});
