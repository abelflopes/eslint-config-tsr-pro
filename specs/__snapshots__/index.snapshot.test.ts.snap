// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`eslint config configuration remains as expected 1`] = `
{
  "env": {
    "browser": true,
    "es2021": true,
    "node": true,
  },
  "extends": [
    "eslint:recommended",
    "plugin:workspaces/recommended",
    "plugin:eslint-comments/recommended",
    "plugin:etc/recommended",
  ],
  "ignorePatterns": [
    "**/dist/**",
    "**/node_modules/*",
    "**/tmp/**",
  ],
  "overrides": [
    {
      "extends": [
        "plugin:jest/recommended",
      ],
      "files": [
        "**/*.{test,spec}.*",
      ],
      "plugins": [
        "jest",
      ],
      "rules": {
        "jest/consistent-test-it": "error",
        "jest/max-expects": "error",
        "jest/max-nested-describe": [
          "error",
          {
            "max": 3,
          },
        ],
        "jest/no-confusing-set-timeout": "warn",
        "jest/no-duplicate-hooks": "error",
        "jest/no-large-snapshots": "warn",
        "jest/no-test-return-statement": "error",
        "jest/no-untyped-mock-factory": "error",
        "jest/prefer-called-with": "warn",
        "jest/prefer-comparison-matcher": "warn",
        "jest/prefer-each": "warn",
        "jest/prefer-equality-matcher": "warn",
        "jest/prefer-hooks-in-order": "error",
        "jest/prefer-hooks-on-top": "error",
        "jest/prefer-lowercase-title": "warn",
        "jest/prefer-mock-promise-shorthand": "warn",
        "jest/prefer-snapshot-hint": "warn",
        "jest/prefer-spy-on": "warn",
        "jest/prefer-strict-equal": "warn",
        "jest/prefer-to-be": "warn",
        "jest/prefer-to-contain": "warn",
        "jest/prefer-to-have-length": "warn",
        "jest/prefer-todo": "warn",
        "jest/require-hook": "error",
        "jest/require-to-throw-message": "warn",
        "jest/require-top-level-describe": "error",
        "jest/unbound-method": "warn",
      },
    },
    {
      "extends": [
        "plugin:@typescript-eslint/recommended-type-checked",
      ],
      "files": [
        "**/*.ts*",
      ],
      "parser": "@typescript-eslint/parser",
      "plugins": [
        "@typescript-eslint",
        "eslint-plugin-tsdoc",
      ],
      "rules": {
        "@typescript-eslint/adjacent-overload-signatures": "error",
        "@typescript-eslint/array-type": [
          "warn",
          {
            "default": "array-simple",
          },
        ],
        "@typescript-eslint/consistent-generic-constructors": "warn",
        "@typescript-eslint/consistent-type-assertions": [
          2,
          {
            "assertionStyle": "never",
          },
        ],
        "@typescript-eslint/consistent-type-exports": [
          "warn",
          {
            "fixMixedExportsWithInlineTypeSpecifier": true,
          },
        ],
        "@typescript-eslint/consistent-type-imports": [
          "warn",
          {
            "fixStyle": "inline-type-imports",
          },
        ],
        "@typescript-eslint/default-param-last": "error",
        "@typescript-eslint/dot-notation": "error",
        "@typescript-eslint/explicit-function-return-type": "error",
        "@typescript-eslint/explicit-member-accessibility": "warn",
        "@typescript-eslint/explicit-module-boundary-types": "warn",
        "@typescript-eslint/init-declarations": "warn",
        "@typescript-eslint/max-params": "error",
        "@typescript-eslint/no-array-delete": "error",
        "@typescript-eslint/no-confusing-non-null-assertion": "error",
        "@typescript-eslint/no-confusing-void-expression": "warn",
        "@typescript-eslint/no-dynamic-delete": "error",
        "@typescript-eslint/no-empty-function": "error",
        "@typescript-eslint/no-empty-interface": "error",
        "@typescript-eslint/no-extraneous-class": "error",
        "@typescript-eslint/no-inferrable-types": "error",
        "@typescript-eslint/no-loop-func": "warn",
        "@typescript-eslint/no-meaningless-void-operator": "error",
        "@typescript-eslint/no-mixed-enums": "warn",
        "@typescript-eslint/no-non-null-asserted-nullish-coalescing": "error",
        "@typescript-eslint/no-non-null-assertion": "error",
        "@typescript-eslint/no-throw-literal": "error",
        "@typescript-eslint/no-unnecessary-boolean-literal-compare": "error",
        "@typescript-eslint/no-unnecessary-condition": "warn",
        "@typescript-eslint/no-unnecessary-qualifier": "error",
        "@typescript-eslint/no-unnecessary-type-arguments": "warn",
        "@typescript-eslint/no-unsafe-unary-minus": "error",
        "@typescript-eslint/no-unused-expressions": "error",
        "@typescript-eslint/no-use-before-define": "error",
        "@typescript-eslint/no-useless-constructor": "error",
        "@typescript-eslint/no-useless-empty-export": "error",
        "@typescript-eslint/no-useless-template-literals": "error",
        "@typescript-eslint/non-nullable-type-assertion-style": "warn",
        "@typescript-eslint/prefer-destructuring": "warn",
        "@typescript-eslint/prefer-enum-initializers": "error",
        "@typescript-eslint/prefer-function-type": "error",
        "@typescript-eslint/prefer-literal-enum-member": "error",
        "@typescript-eslint/prefer-optional-chain": "warn",
        "@typescript-eslint/prefer-promise-reject-errors": "error",
        "@typescript-eslint/prefer-readonly": "warn",
        "@typescript-eslint/prefer-reduce-type-parameter": "error",
        "@typescript-eslint/prefer-string-starts-ends-with": "warn",
        "@typescript-eslint/promise-function-async": "error",
        "@typescript-eslint/require-array-sort-compare": "error",
        "@typescript-eslint/return-await": "warn",
        "@typescript-eslint/switch-exhaustiveness-check": "error",
        "@typescript-eslint/unified-signatures": "warn",
        "default-param-last": "off",
        "dot-notation": "off",
        "init-declarations": "off",
        "max-params": "off",
        "no-empty-function": "off",
        "no-loop-func": "off",
        "no-throw-literal": "off",
        "no-unused-expressions": "off",
        "no-use-before-define": "off",
        "no-useless-constructor": "off",
        "prefer-destructuring": "off",
        "prefer-promise-reject-errors": "off",
        "tsdoc/syntax": "warn",
      },
    },
    {
      "extends": [
        "plugin:react/recommended",
        "plugin:react-hooks/recommended",
        "plugin:jsx-a11y/recommended",
      ],
      "files": [
        "**/*.{tsx,jsx}",
      ],
      "parserOptions": {
        "ecmaFeatures": {
          "jsx": true,
        },
      },
      "plugins": [
        "react",
        "react-hooks",
        "react-refresh",
        "jsx-a11y",
      ],
      "rules": {
        "jsx-a11y/anchor-ambiguous-text": "warn",
        "jsx-a11y/control-has-associated-label": "warn",
        "jsx-a11y/lang": "warn",
        "jsx-a11y/no-aria-hidden-on-focusable": "warn",
        "jsx-a11y/prefer-tag-over-role": "warn",
        "react-refresh/only-export-components": "warn",
        "react/boolean-prop-naming": "warn",
        "react/function-component-definition": [
          "warn",
          {
            "namedComponents": "arrow-function",
          },
        ],
        "react/hook-use-state": [
          "error",
          {
            "allowDestructuredState": true,
          },
        ],
        "react/iframe-missing-sandbox": "warn",
        "react/jsx-boolean-value": "warn",
        "react/jsx-closing-tag-location": "warn",
        "react/jsx-curly-spacing": "warn",
        "react/jsx-equals-spacing": "warn",
        "react/jsx-filename-extension": [
          "warn",
          {
            "extensions": [
              ".jsx",
              ".tsx",
            ],
          },
        ],
        "react/jsx-first-prop-new-line": "warn",
        "react/jsx-fragments": "warn",
        "react/jsx-handler-names": "warn",
        "react/jsx-max-depth": [
          "warn",
          {
            "max": 4,
          },
        ],
        "react/jsx-no-constructed-context-values": "error",
        "react/jsx-no-leaked-render": "warn",
        "react/jsx-no-useless-fragment": "error",
        "react/jsx-pascal-case": "warn",
        "react/jsx-props-no-multi-spaces": "warn",
        "react/jsx-sort-props": [
          "warn",
          {
            "callbacksLast": true,
            "ignoreCase": true,
            "locale": "auto",
            "multiline": "last",
            "noSortAlphabetically": true,
            "reservedFirst": true,
            "shorthandFirst": false,
            "shorthandLast": true,
          },
        ],
        "react/jsx-tag-spacing": "warn",
        "react/jsx-wrap-multilines": "warn",
        "react/no-array-index-key": "error",
        "react/no-danger": "warn",
        "react/no-invalid-html-attribute": "warn",
        "react/no-multi-comp": "error",
        "react/no-namespace": "warn",
        "react/no-object-type-as-default-prop": "error",
        "react/no-this-in-sfc": "error",
        "react/no-typos": "error",
        "react/no-unstable-nested-components": "error",
        "react/prefer-read-only-props": "error",
        "react/prop-types": "off",
        "react/self-closing-comp": "warn",
        "react/void-dom-elements-no-children": "error",
      },
      "settings": {
        "react": {
          "version": "detect",
        },
      },
    },
  ],
  "parserOptions": {
    "ecmaVersion": 2015,
    "project": true,
    "sourceType": "module",
  },
  "plugins": [
    "prettier",
    "import",
    "eslint-comments",
  ],
  "rules": {
    "array-callback-return": "error",
    "arrow-body-style": [
      "error",
      "as-needed",
    ],
    "block-scoped-var": "error",
    "class-methods-use-this": "warn",
    "complexity": [
      "warn",
      10,
    ],
    "consistent-this": "warn",
    "curly": [
      "error",
      "multi-or-nest",
      "consistent",
    ],
    "default-case": "warn",
    "default-case-last": "error",
    "default-param-last": "error",
    "dot-notation": "error",
    "eqeqeq": "error",
    "eslint-comments/no-unused-disable": "warn",
    "eslint-comments/require-description": [
      "warn",
      {
        "ignore": [
          "eslint-enable",
        ],
      },
    ],
    "etc/no-commented-out-code": "warn",
    "etc/no-const-enum": "warn",
    "etc/no-enum": "warn",
    "etc/throw-error": "error",
    "etc/underscore-internal": "warn",
    "grouped-accessor-pairs": "error",
    "import/consistent-type-specifier-style": [
      "warn",
      "prefer-inline",
    ],
    "import/export": "warn",
    "import/exports-last": "warn",
    "import/first": "error",
    "import/named": "warn",
    "import/newline-after-import": "warn",
    "import/no-absolute-path": "error",
    "import/no-cycle": "error",
    "import/no-deprecated": "error",
    "import/no-duplicates": "warn",
    "import/no-empty-named-blocks": "error",
    "import/no-extraneous-dependencies": "error",
    "import/no-mutable-exports": "warn",
    "import/no-named-as-default": "warn",
    "import/no-named-as-default-member": "warn",
    "import/no-relative-packages": "error",
    "import/no-relative-parent-imports": "warn",
    "import/no-self-import": "error",
    "import/no-unused-modules": "warn",
    "import/no-useless-path-segments": "error",
    "import/no-webpack-loader-syntax": "warn",
    "init-declarations": "warn",
    "line-comment-position": [
      "warn",
      {
        "position": "above",
      },
    ],
    "max-classes-per-file": "error",
    "max-depth": [
      "warn",
      4,
    ],
    "max-len": [
      2,
      {
        "code": 100,
        "ignoreComments": true,
        "ignoreRegExpLiterals": true,
        "ignoreStrings": true,
        "ignoreTemplateLiterals": true,
      },
    ],
    "max-lines": [
      "warn",
      200,
    ],
    "max-lines-per-function": [
      "warn",
      150,
    ],
    "max-nested-callbacks": [
      "warn",
      5,
    ],
    "max-params": "error",
    "new-cap": "warn",
    "no-alert": "error",
    "no-await-in-loop": "error",
    "no-bitwise": "error",
    "no-caller": "error",
    "no-console": "error",
    "no-constant-binary-expression": "error",
    "no-duplicate-imports": "error",
    "no-else-return": "error",
    "no-empty-function": "error",
    "no-empty-static-block": "error",
    "no-eq-null": "error",
    "no-eval": "error",
    "no-extra-bind": "error",
    "no-extra-label": "error",
    "no-implicit-coercion": "warn",
    "no-implicit-globals": "error",
    "no-implied-eval": "error",
    "no-invalid-this": "error",
    "no-iterator": "error",
    "no-label-var": "error",
    "no-labels": "error",
    "no-lone-blocks": "error",
    "no-lonely-if": "error",
    "no-loop-func": "warn",
    "no-multi-assign": "warn",
    "no-negated-condition": "warn",
    "no-nested-ternary": "warn",
    "no-new": "error",
    "no-new-func": "error",
    "no-new-native-nonconstructor": "error",
    "no-new-wrappers": "error",
    "no-object-constructor": "error",
    "no-param-reassign": "error",
    "no-plusplus": "error",
    "no-promise-executor-return": "error",
    "no-proto": "error",
    "no-restricted-exports": "error",
    "no-script-url": "error",
    "no-self-compare": "error",
    "no-template-curly-in-string": "error",
    "no-throw-literal": "error",
    "no-underscore-dangle": "error",
    "no-unmodified-loop-condition": "error",
    "no-unneeded-ternary": "warn",
    "no-unreachable-loop": "error",
    "no-unused-expressions": "error",
    "no-unused-private-class-members": "error",
    "no-use-before-define": "error",
    "no-useless-call": "error",
    "no-useless-computed-key": "error",
    "no-useless-concat": "error",
    "no-useless-constructor": "error",
    "no-useless-rename": "error",
    "no-useless-return": "error",
    "no-var": "error",
    "no-void": [
      "error",
      {
        "allowAsStatement": true,
      },
    ],
    "object-shorthand": [
      "warn",
      "properties",
    ],
    "operator-assignment": "warn",
    "padding-line-between-statements": [
      "error",
      {
        "blankLine": "always",
        "next": "export",
        "prev": "*",
      },
      {
        "blankLine": "always",
        "next": "*",
        "prev": "export",
      },
      {
        "blankLine": "always",
        "next": "try",
        "prev": "*",
      },
    ],
    "prefer-arrow-callback": "warn",
    "prefer-const": "error",
    "prefer-destructuring": "warn",
    "prefer-object-has-own": "warn",
    "prefer-object-spread": "warn",
    "prefer-promise-reject-errors": "error",
    "prefer-rest-params": "warn",
    "prefer-spread": "warn",
    "prefer-template": "warn",
    "prettier/prettier": "warn",
    "radix": [
      "error",
      "as-needed",
    ],
    "require-atomic-updates": "error",
    "require-unicode-regexp": "warn",
    "sort-imports": [
      "warn",
      {
        "allowSeparatedGroups": true,
        "ignoreDeclarationSort": true,
        "ignoreMemberSort": true,
      },
    ],
    "symbol-description": "error",
    "vars-on-top": "error",
    "yoda": "warn",
  },
}
`;
